### PR TITLE
[WIP] add delete provider capability

### LINKED ISSUE #342: Be able to delete a provider
Do not delete the provider but set the field `deleted` to True.
If the field `deleted` is not present in the model, create it.

### PR DESCRIPTION
**Pull request by boney-bun**
fix #342

### CODE REVIEW

**Comment by boney-bun** [comment]
PIA:
ford3#398

questions:
1. i define red style button for the `delete button`. what do you think @pierrealixt ?
2. so we don't want to do hard delete. right?
3. my migration file contains `saqaqualification.name`. definitely it's not come from my code. does someone forget to do migration before or my local is dirty?

File: django_project/ford3/views/views.py [hunk 1]
```diff
@@ -34,6 +40,11 @@ def show_qualification(request, provider_id, campus_id, qualification_id):
     qualification = get_object_or_404(
         Qualification,
         id=qualification_id)
+
+    # prevent a knowledgeable user looks at deleted provider
+    if qualification.campus.provider.deleted:
```
Comment (pierrealixt) [inline django_project/ford3/views/views.py]: you repeated this code 6 times. Please refactor

**Comment by pierrealixt** [comment]
@boney-bun I'm not sure about the UI. I don't think this button will be on the provider page. I don't think it should redirect the user back to the landing page.

File: django_project/ford3/urls.py [hunk 2]
```diff
@@ -52,6 +51,9 @@
     path('providers/<int:provider_id>/edit',
          provider.edit,
          name='edit-provider'),
+    path('providers/<int:provider_id>/remove',
```
Comment (pierrealixt) [inline django_project/ford3/urls.py]: we don't need this for now

File: django_project/ford3/views/provider.py [hunk 3]
```diff
@@ -136,3 +138,15 @@ def show(request, provider_id):
         context['provider_logo'] = provider.provider_logo.url
 
     return render(request, 'provider.html', context)
+
+
+def remove(request, provider_id):
```
Comment (pierrealixt) [inline django_project/ford3/views/provider.py]: we don't need this for now

File: django_project/ford3/templates/provider.html [hunk 4]
```diff
@@ -7,11 +7,16 @@
 {% block body %}
 	<div class="container mt-3">
 		<div class="row no-gutters">
-			<div class='col-md-9'>
+			<div class='col-md-7'>
 				<h3>{{ provider.name }}</h3>
 				<h4>Campuses</h4>
 				<hr />
 			</div>
+            <div class='col-md-2 px-2'>
```
Comment (pierrealixt) [inline django_project/ford3/templates/provider.html]: we don't need this for now

File: django_project/ford3/tests/views/test_provider.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
+from django.urls import reverse
+from django.test import TestCase
+from ford3.tests.models.model_factories import ModelFactories
+
+
+class TestProvider(TestCase):
```
Comment (pierrealixt) [inline django_project/ford3/tests/views/test_provider.py]: add a model test instead

Comment (boney-bun) [reply django_project/ford3/tests/views/test_provider.py]: remove test case is elimated. 
however, i'll like adding add and edit test cases for provider in this PR.
or, do you prefer having it in separate PR?

File: django_project/ford3/views/saqa_qualifications.py [hunk 6]
```diff
@@ -1,13 +1,18 @@
...
+from django.shortcuts import get_object_or_404
+from ford3.models import (
+    Provider,
+    SAQAQualification
+)
 
 
 def json_response(results):
-    return JsonResponse({'results': results})
+    return HttpResponse(
```
Comment (pierrealixt) [inline django_project/ford3/views/saqa_qualifications.py:12]: why did you do this?

**Pull request closed without merging**